contracts/utils/Whitelist.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.0 <0.8.0;
+
+import "./AccessControl.sol";
+
+contract Whitelist is AccessControl {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    event WhitelistEnabled(address sender);
+    event WhitelistDisabled(address sender);
+    event WhitelistAccountAdded(address account);
+    event WhitelistAccountRemoved(address account);
+
+    bytes32 public constant WHITELIST_ROLE = keccak256("WHITELIST_ROLE");
+
+    bool private _isWhitelistEnabled;
+    EnumerableSet.AddressSet private _whitelist;
+
+    modifier onlyWhitelistOrAdmin() {
+        _onlyWhitelistOrAdmin();
+        _;
+    }
+
+    modifier usingWhitelist() {
+        _usingWhitelist();
+        _;
+    }
+
+    function enableWhitelist() external onlyWhitelistOrAdmin {
```

Comment:
Add comments with documentation to all non-private functions

Reply:
In progress...

Reply:
[done](https://github.com/Cerebellum-Network/chainbridge-solidity/pull/4/files#diff-d1d675d04151a0a3ec00bad5e7e2fbea4c4d2e70a75a00f656df8fa08553f5f7R64-R68)